### ISSUE TITLE
Ability to Search for components on Documentation site

### ISSUE THREAD

**Issue by jeremysmartt** [Teradata/covalent#415]
This repository's issues are reserved for feature requests and bug reports.

#### Feature Request

As the Covalent Features grow it would be nice to make it easier to search for components on the documentation website.

#### What is the expected behavior?

Ability to search by free text or autocomplete or both and find components on the documentation site.

#### What is the motivation / use case for changing the behavior?
Ease of finding components

**Comment by kyleledbetter** [comment]
Started on a branch for this https://github.com/Teradata/covalent/tree/feature/autocomplete-search

![autocomplete-search](https://cloud.githubusercontent.com/assets/867883/23771180/a072144c-04db-11e7-8113-d9b44c05e07b.gif)

**Comment by jdrew1303** [comment]
At the moment the design uses the autocomplete box. Would it be possible to have it use the existing search component with an autocomplete dropdown? This is similar to:
![image](https://cloud.githubusercontent.com/assets/2535432/25729698/eaa73e96-312e-11e7-945e-b6e7a2eca04c.png)

**Comment by emoralesb05** [comment]
@jdrew1303 yep, thats a feature we are thinking to add soon. `md-autocomplete` usage in `td-search-box` and `td-search-input`

**Comment by jdrew1303** [comment]
Cool. Cant wait for this to land!
